plio/views.py

```diff
@@ -451,79 +451,94 @@ def metrics(self, request, uuid):
                     "item_type",
                     "question_type",
                     "correct_answer",
+                    "survey",
                 ],
             )
 
             # retain only the responses to items which are questions
-            question_df = df[df["item_type"] == "question"].reset_index(drop=True)
-            num_questions = len(questions)
-
-            def is_answer_correct(row):
-                """Whether the answer corresponding to the given row is correct"""
-                if row["question_type"] in ["mcq", "checkbox"]:
-                    return row["answer"] == row["correct_answer"]
-                return row["answer"] is not None
-
-            # holds the number of questions answered for each viewer
-            num_answered_list = []
-            # holds the number of questions correctly answered for each viewer
-            num_correct_list = []
-
-            user_grouping = question_df.groupby("user_id")
-            for group in user_grouping.groups:
-                # get the responses for a given user
-                group_df = user_grouping.get_group(group)
-
-                # sanity check
-                assert num_questions == len(
-                    group_df
-                ), "Inconsistency in the number of questions"
-
-                num_answered = sum(
-                    group_df["answer"].apply(lambda value: value is not None)
+            dff = df[df["item_type"] == "question"].reset_index(drop=True)
```

Comment:
why create a separate data frame? you can simply update the `question_df` right?